apps/proxy/src/app/create-project/create-project.store.ts

```diff
@@ -0,0 +1,158 @@
+import { Router } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { ComponentStore, tapResponse } from '@ngrx/component-store';
+import { IEnvironments, IProjects } from '@proxy/models/logs-models';
+import { BaseResponse, IPaginatedResponse, IReqParams, errorResolver } from '@proxy/models/root-models';
+import { CreateProjectService } from '@proxy/services/proxy/create-project';
+
+import { PrimeNgToastService } from '@proxy/ui/prime-ng-toast';
+
+import { EMPTY, Observable, catchError, switchMap } from 'rxjs';
+import { ICreateSource } from '@proxy/models/project-model';
+
+export interface ICreateProjectInitialState {
+    projects: IPaginatedResponse<IProjects[]>;
+    environments: IPaginatedResponse<IEnvironments[]>;
+    sourceDomain: any;
+    isLoading: boolean;
+    getProject: boolean;
+}
+
+@Injectable()
+export class CreateProjectComponentStore extends ComponentStore<ICreateProjectInitialState> {
+    constructor(private service: CreateProjectService, private toast: PrimeNgToastService, private router: Router) {
+        super({
+            projects: null,
+            environments: null,
+            sourceDomain: null,
+            isLoading: false,
+            getProject: false,
+        });
+    }
+
+    readonly projects$: Observable<IPaginatedResponse<IProjects[]>> = this.select((state) => state.projects);
+    readonly environments$: Observable<IPaginatedResponse<IEnvironments[]>> = this.select(
+        (state) => state.environments
+    );
+    readonly sourceDomain$: Observable<any> = this.select((state) => state.sourceDomain);
+    readonly getProject$: Observable<boolean> = this.select((state) => state.getProject);
+    readonly isLoading$: Observable<boolean> = this.select((state) => state.isLoading);
+
+    readonly getEnvironment = this.effect((data: Observable<IReqParams>) => {
+        return data.pipe(
+            switchMap((req) => {
+                return this.service.getEnvironments(req).pipe(
+                    tapResponse(
+                        (res: BaseResponse<IPaginatedResponse<IEnvironments[]>, void>) => {
+                            if (res.hasError) {
+                                this.showError(res.errors);
+                            }
+                            return this.patchState((state) => ({
+                                environmentsInProcess: false,
+                                environments:
+                                    res?.data?.pageNo > 1
+                                        ? { ...res.data, data: [...state.environments.data, ...res.data.data] }
```

Comment:
If we are not using virtual scroll for environment dropdown, why are we managing pagination.